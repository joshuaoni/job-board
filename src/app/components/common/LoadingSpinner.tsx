interface LoadingSpinnerProps {
  size?: "small" | "medium" | "large";
  className?: string;
}

const sizeClasses = {
  small: "w-4 h-4",
  medium: "w-8 h-8",
  large: "w-12 h-12",
};

export const LoadingSpinner = ({
  size = "medium",
  className = "",
}: LoadingSpinnerProps) => {
  return (
    <div
      className={`inline-block animate-spin rounded-full border-2 border-solid border-current border-r-transparent motion-reduce:animate-[spin_1.5s_linear_infinite] ${sizeClasses[size]} ${className}`}
      role="status"
      aria-label="Loading..."
    >
      <span className="sr-only">Loading...</span>
    </div>
  );
};
