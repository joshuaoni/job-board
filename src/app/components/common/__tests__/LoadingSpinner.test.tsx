import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { LoadingSpinner } from "../LoadingSpinner";

describe("LoadingSpinner", () => {
  it("should render with default size", () => {
    render(<LoadingSpinner />);
    const spinner = screen.getByRole("status");
    expect(spinner).toBeInTheDocument();
    expect(spinner).toHaveClass("w-8", "h-8"); // medium is default
  });

  it("should render with small size", () => {
    render(<LoadingSpinner size="small" />);
    const spinner = screen.getByRole("status");
    expect(spinner).toBeInTheDocument();
    expect(spinner).toHaveClass("w-4", "h-4");
  });

  it("should render with large size", () => {
    render(<LoadingSpinner size="large" />);
    const spinner = screen.getByRole("status");
    expect(spinner).toBeInTheDocument();
    expect(spinner).toHaveClass("w-12", "h-12");
  });

  it("should have correct ARIA attributes", () => {
    render(<LoadingSpinner />);
    const spinner = screen.getByRole("status");
    expect(spinner).toHaveAttribute("aria-label", "Loading...");
  });

  it("should have animation classes", () => {
    render(<LoadingSpinner />);
    const spinner = screen.getByRole("status");
    expect(spinner).toHaveClass("animate-spin");
  });

  it("should accept custom className", () => {
    render(<LoadingSpinner className="test-class" />);
    const spinner = screen.getByRole("status");
    expect(spinner).toHaveClass("test-class");
  });
});
