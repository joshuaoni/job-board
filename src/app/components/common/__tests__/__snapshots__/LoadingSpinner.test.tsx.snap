// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner should match snapshot with custom className 1`] = `
<div>
  <div
    aria-label="Loading..."
    class="inline-block animate-spin rounded-full border-2 border-solid border-current border-r-transparent motion-reduce:animate-[spin_1.5s_linear_infinite] w-8 h-8 test-class"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner should match snapshot with default size 1`] = `
<div>
  <div
    aria-label="Loading..."
    class="inline-block animate-spin rounded-full border-2 border-solid border-current border-r-transparent motion-reduce:animate-[spin_1.5s_linear_infinite] w-8 h-8 "
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner should match snapshot with large size 1`] = `
<div>
  <div
    aria-label="Loading..."
    class="inline-block animate-spin rounded-full border-2 border-solid border-current border-r-transparent motion-reduce:animate-[spin_1.5s_linear_infinite] w-12 h-12 "
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`LoadingSpinner should match snapshot with small size 1`] = `
<div>
  <div
    aria-label="Loading..."
    class="inline-block animate-spin rounded-full border-2 border-solid border-current border-r-transparent motion-reduce:animate-[spin_1.5s_linear_infinite] w-4 h-4 "
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
